rplugin/python3/deoplete/sources/deoplete_jedi/worker.py

```diff
@@ -62,13 +62,10 @@ def completion_work(self, cache_key, extra_modules, source, line, col,
     def run(self):
         while True:
             try:
-                work = self.in_queue.get(block=False, timeout=0.5)
+                work = self.in_queue.get()
                 self.log.debug('Got work')
-                self.out_queue.put(self.completion_work(*work), block=False)
+                self.out_queue.put(self.completion_work(*work), timeout=0.5)
                 self.log.debug('Completed work')
-            except queue.Empty:
-                # Sleep is mandatory to avoid pegging the CPU
-                time.sleep(0.01)
             except Exception:
                 self.log.debug('Worker error', exc_info=True)
                 time.sleep(0.05)
```

Comment:
I guess this sleep can be removed since the queue blocks.  I think this was added for sanity to keep a bad thread from exhausting the queue, but if one goes bad any other threads will probably follow suit anyways.